### ISSUE TITLE
nodeHandle expected to be non-null

### ISSUE THREAD

**Issue by Zeneixe** [react-native-community/react-native-webview#908]
Error raised:
`nodeHandle expected to be non-null`
Upon calling `goBack();` on the webview.

This is happening on our production app and we haven't found a repro case yet. It does seem to occur only on iOS though.

We're using RN version 0.59 (latest), rn webview version 5.10.0.

Any ideas what could be causing it?

**Comment by ErangaD** [comment]
I am experiencing this as well. I am using **React Navigation** for my app. In the initial load of the web view screen, `goBack()` works fine. But when I go to another screen and come back to the screen containing the web view, the error `nodeHandle expected to be non-null` gets raised. I am trying to dynamically changing the `source={{uri: currentURL}}` url to be loaded here. Could that be a reason?

**Comment by ErangaD** [comment]
> I am experiencing this as well. I am using **React Navigation** for my app. In the initial load of the web view screen, `goBack()` works fine. But when I go to another screen and come back to the screen containing the web view, the error `nodeHandle expected to be non-null` gets raised. I am trying to dynamically changing the `source={{uri: currentURL}}` url to be loaded here. Could that be a reason?

In my case, I had kept a global reference for the `ref` prop of the WebView. By removing that with a class level reference, the error was fixed.

**Comment by Zeneixe** [comment]
Thanks for the reply! Can you be more specific maybe showing the before/after code?

Currently it looks like this for us:

`ref={(webView: WebView): void => { this.webView = webView; }}`

**Comment by liusanhong** [comment]
me too

**Comment by Zeneixe** [comment]
Bump

**Comment by Roxk-33** [comment]
me too

**Comment by vincentnavetat** [comment]
me too

**Comment by bryvin** [comment]
Any update on this? We are also experiencing this issue.

**Comment by 1797918430** [comment]
I got the same error ,it's may by due to low version of react-native ,but i update react-native fail
finally i fix it  by replace function below:
   _this.getCommands = function () { return  UIManager.RNCWebView.Commands; };

 _this.getWebViewHandle = function () {
            var nodeHandle = findNodeHandle(_this.webViewRef.value);
            invariant(nodeHandle != null, 'nodeHandle expected to be non-null');
            return nodeHandle;
        };

by the way ,sorry for my bad english

**Comment by aperomingo** [comment]
This bots...

As the issue is clearly not resolved and also I have the problem too, I found a solution, my solution at leasts.
I had this piece code:
```
<WebView
     ref={this._setWebviewRef}
     onNavigationStateChange={this._handleWebViewNavigationStateChange}
     onMessage={this._handleOnMessage}
     source={{uri: this.state.dashboardUrl || ''}}
     renderLoading={() => <LoadingSpinner />}
     startInLoadingState
     incognito
/>
```
where `this._setWebviewRef` was the next:
```
_setWebviewRef = (ref) => {
    if (ref && !this.props.webviewRef) {
      const {goBack} = ref;
      goBack && this.props.setWebviewRef({goBack});
    }
};
```
I'm using `redux` to save `goBack` method to use it on the typic `_handleAndroidBackButton`. So as other guys said, i was giving the fu***** `nodeHandle expected to be non-null` but i was seeing that goBack method exists on `_handleAndroidBackButton` context.
Debugging i saw that `ref` WebView method has multiple calls, not just 1 time as i was controlling. So, deleting this condition `&& !this.props.webviewRef` is already working.

Also, I had tried to set my uri 'hardcoded' with the same error, so that not work for me.
